File: Java/src/main/java/com/xzq/java/genericity/gener/BlueColor.java
package com.xzq.java.genericity.gener;

public class BlueColor extends Color<Blue> {

    public BlueColor(Blue blue) {

    }

    @Override
    void priceColor() {

    }

    public static <T> T fromJson(String json, Class<T> tClass) {
        T t = null;
        try {
            t = tClass.getDeclaredConstructor().newInstance();
        } catch (Exception e) {
            e.printStackTrace();
        }
        return t;
    }
}

File: JavaStudy/src/com/xzq/java/genericity/genkt/BlueColor.kt
package com.xzq.java.genericity.genkt

class BlueColor(t: Blue) : Color<Blue>(t) {

    override fun priceColor() {

    }

    fun <T> fromJson(json: String, clazz: Class<T>): T? {
        return clazz.getDeclaredConstructor().newInstance()
    }


    //多个上界的情况
    fun <T> test(list: List<T>, threshold: T): List<T> where T : CharSequence, T : Comparable<T> {
        return list.filter { it > threshold }.map { it }
    }

}